src/pf-popover/pf-popover.component.js

```diff
@@ -0,0 +1,421 @@
+  */
+  get targetSelector() {
+    return this._targetSelector;
+  }
+
+  /**
+   * Set targetSelector
+   *
+   * @param {string} value The target element selector
+   */
+  set targetSelector(value) {
+    if (this._targetSelector !== value) {
+      this._targetSelector = value;
+      this._target = document.querySelector(this._targetSelector);
+      this.setAttribute('targetSelector', value);
+    }
+  }
+
+  /**
+   * Get the popoverTitle
+   *
+   * @return {string} the title of popover
+   */
+  get popoverTitle() {
+    return this._popoverTitle;
+  }
+
+  /**
+   * Set popoverTitle
+   *
+   * @param {string} value The title of popover
+   */
+  set popoverTitle(value) {
+    if (this._popoverTitle !== value) {
+      this._popoverTitle = value;
+      this.setAttribute('popoverTitle', value);
+    }
+  }
+
+  /**
+     * The popover open method
+     */
+  open() {
+    clearTimeout(this._timer);
+    this._timer = setTimeout(() => {
+      if (this.popover === null) {
+        this._createPopover();
+        this._stylePopover();
+        this._checkPlacement();
+        this._showPopover();
+        //notify frameworks
+        this.dispatchEvent(new CustomEvent('pf-popover.opened', {}));
+      }
+    }, 20);
+  }
+
+  /**
+   * The popover close method
+   */
+  close() {
+    clearTimeout(this._timer);
+    this._timer = setTimeout(() => {
+      if (this.popover && this.popover !== null) {
+        pfUtil.removeClass(this.popover, 'in');
+        setTimeout(() => {
+          this._removePopover();
+          //notify frameworks
+          this.dispatchEvent(new CustomEvent('pf-popover.closed', {}));
+        }, this._duration);
+      }
+    }, this._delay + this._duration);
+  }
+
+  /**
+   * Removes the popover
+   * @private
+   */
+  _removePopover() {
+    this.popover && this._container.removeChild(this.popover);
+    this.popover = null;
+  }
+
+  /**
+   * Creates the popover
+   * @private
+   */
+  _createPopover() {
+    let clone = document.importNode(this._template.content, true);
+    let popoverInner = clone.querySelector('.popover-content');
+    let popovertitle = clone.querySelector('.popover-title');
+    let closeButton = document.createElement('template');
+    closeButton.innerHTML = `<button type="button" class="close"><span class="pficon pficon-close"></span></button>`;
+
+    if (this._popoverTitle === '' && !this._dismissible) {
+      popovertitle.parentNode.removeChild(popovertitle);
+    } else {
+      //set popover title
+      popovertitle.innerHTML = this._popoverTitle;
+
+      if (this._dismissible) {
+        popovertitle.appendChild(closeButton.content);
+      }
+    }
+    //set popover content
+    popoverInner.innerHTML = this.content;
+
+    //append to the container
+    this._container.appendChild(clone);
+
+    //set reference to appended node
+    let popovers = this._container.querySelectorAll('.popover');
+    this.popover = popovers[popovers.length - 1];
```

Comment:
Not a requirement... Having trouble understanding if there will ever be more than one ```.popover``` element, but if it fits into the concept could try.... 

less specific way to handle this using ```:first-child```
```javascript
clone = clone.querySelector(':first-child');
this._container.appendChild(clone);
this.popover = clone;
```